scripts/nmt/s2s_mic.py

```diff
@@ -0,0 +1,118 @@
+    parser = argparse.ArgumentParser(
+        description="Streaming speech to speech translation from microphone via Riva AI Services",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("--input-device", type=int, default=default_device_index, help="An input audio device to use.")
+    parser.add_argument("--list-input-devices", action="store_true", help="List input audio device indices.")
+    parser.add_argument("--list-output-devices", action="store_true", help="List input audio device indices.")
+    parser.add_argument("--output-device", type=int, help="Output device to use.")
+    parser.add_argument("--target-language-code", default="en-US", help="Language code of the output language.")
+    parser.add_argument(
+        "--play-audio",
+        action="store_true",
+        help="Play input audio simultaneously with transcribing and translating it. If `--output-device` is not provided, "
+        "then the default output audio device will be used.",
+    )
+
+    parser = add_asr_config_argparse_parameters(parser, profanity_filter=True)
+    parser = add_connection_argparse_parameters(parser)
+    parser.add_argument(
+        "--sample-rate-hz",
+        type=int,
+        help="A number of frames per second in audio streamed from a microphone.",
+        default=16000,
+    )
+    parser.add_argument(
+        "--file-streaming-chunk",
+        type=int,
+        default=1600,
+        help="A maximum number of frames in a audio chunk sent to server.",
+    )
+    args = parser.parse_args()
+    return args
+
+def play_responses(responses: Iterable[riva_nmt.StreamingTranslateSpeechToSpeechResponse],
+                   sound_stream) -> None:
+    count = 0
+    for response in responses:
+        #if first:
+            #print(f"time to first audio {(stop - start):.3f}s")
+        #    first=False
+        if sound_stream is not None:
+            sound_stream(response.speech.audio)
+            fname = "response" + str(count)
+            out_f = wave.open(fname, 'wb')
+            out_f.setnchannels(1)
+            out_f.setsampwidth(2)
+            out_f.setframerate(44100)
+        count += 1
+
+
+def main() -> None:
+    args = parse_args()
+    sound_stream = None
+    sampwidth = 2
+    nchannels = 1
+    if args.list_input_devices:
+        riva.client.audio_io.list_input_devices()
+        return
+    if args.output_device is not None or args.play_audio:
+        print("playing audio")
+        sound_stream = riva.client.audio_io.SoundCallBack(
+            args.output_device, nchannels=nchannels, sampwidth=sampwidth, framerate=44100
+        )
+        print(sound_stream)
+    first = True # first tts output chunk received
+    auth = riva.client.Auth(args.ssl_cert, args.use_ssl, args.server)
+    nmt_service = riva.client.NeuralMachineTranslationClient(auth)
+    s2s_config = riva.client.StreamingTranslateSpeechToSpeechConfig(
```

Comment:
Do we need a `tts_config` as in [proto](https://github.com/nvidia-riva/common/blob/4105452f13ae96be1b15db48d97040ac79a3a509/riva/proto/riva_nmt.proto#L46)? If so, then we could add a `add_tts_config_argparse_parameters()` function to `argparse_utils.py` function and refactor `tts/talk.py` using this function.